test/tests/functional/pbs_node_rampdown.py

```diff
@@ -6362,3 +6415,774 @@ def test_release_nodes_shared_server_restart_immed(self):
+                            {'resources_available.ncpus': 3},
+                            id=self.hostC)
+
+        # Submit multiple jobs
+        jid1 = self.create_and_submit_job('job13')
+        jid2 = self.create_and_submit_job('job13')
+        jid3 = self.create_and_submit_job('job13')
+
+        e_host_j1 = self.hostA + "/0+" + self.hostB + "/0+" + self.hostC + "/0"
+        e_host_j2 = self.hostA + "/1+" + self.hostB + "/1+" + self.hostC + "/1"
+        e_host_j3 = self.hostA + "/2+" + self.hostB + "/2+" + self.hostC + "/2"
+        e_vnode = "(%s:ncpus=1)+(%s:ncpus=1)+(%s:ncpus=1)" \
+            % (self.hostA, self.hostB, self.hostC)
+
+        self.server.expect(JOB, {"job_state=R": 3})
+        self.server.expect(JOB, {"exec_host": e_host_j1,
+                                 "exec_vnode": e_vnode}, id=jid1)
+        self.server.expect(JOB, {"exec_host": e_host_j2,
+                                 "exec_vnode": e_vnode}, id=jid2)
+        self.server.expect(JOB, {"exec_host": e_host_j3,
+                                 "exec_vnode": e_vnode}, id=jid3)
+
+        # Verify that 3 processes running on hostB
+        process = 0
+        self.server.pu.get_proc_info(
+            self.momB.hostname, ".*fib.*", None, regexp=True)
+        if (self.server.pu.processes is not None):
+            for key in self.server.pu.processes:
+                if ("fib" in key):
+                    process = len(self.server.pu.processes[key])
+                    print "length of the process is %d " % (process,)
+
+        self.assertEqual(process, 3)
+
+        # Release node2 from job1 only
+        cmd = [self.pbs_release_nodes_cmd, '-j', jid1, self.hostB]
+        ret = self.server.du.run_cmd(self.server.hostname, cmd,
+                                     runas=TEST_USER)
+        self.assertEqual(ret['rc'], 0)
+
+        self.momB.log_match("Job;%s;DELETE_JOB2 received" % (jid1,),
+                            max_attempts=18, interval=2)
+
+        # Verify that only 2 process left on hostB now
+        process = 0
+        self.server.pu.get_proc_info(
+            self.momB.hostname, ".*fib.*", None, regexp=True)
+        if (self.server.pu.processes is not None):
+            for key in self.server.pu.processes:
+                if ("fib" in key):
+                    process = len(self.server.pu.processes[key])
+                    print "length of the process is %d " % (process,)
+        self.assertEqual(process, 2)
+
+        # Mom logs only have message for job1 for node3
+        self.momA.log_match(
+            "Job;%s;%s.+cput=.+mem.+" % (jid1, self.hostB),
+            max_attempts=18, interval=2, regexp=True)
+
+        self.momA.log_match(
+            "Job;%s;%s.+cput=.+mem.+" % (jid2, self.hostB),
+            max_attempts=8, regexp=True,
+            existence=False)
+
+        self.momA.log_match(
+            "Job;%s;%s.+cput=.+mem.+" % (jid3, self.hostB),
+            max_attempts=5, regexp=True,
```

Comment:
Why the difference in max_attempts (8 vs 5) ?

Reply:
can't recall. May be just want to make sure it is not there as things on VMs are slower